src/workerd/util/mimetype.c++

```diff
@@ -6,24 +6,64 @@
 #include <kj/debug.h>
 #include <kj/string-tree.h>
 #include <workerd/util/string-buffer.h>
+#include <array>
 
 namespace workerd {
 
 namespace {
 
-bool isWhitespace(const char c) {
+constexpr bool isWhitespace(const char c) noexcept {
   return (c == '\r' || c == '\n' || c == '\t' || c == ' ');
 }
 
-bool isTokenChar(const unsigned char c) {
-  return c == '!' || c == '#' || c == '$' || c == '%' || c == '&' || c == '\'' ||
-         c == '*' || c == '+' || c == '\\' || c == '-' || c == '.' || c == '^' ||
-         c == '_' || c == '`' || c == '|' || c == '~' || (c >= 'A' && c <= 'Z') ||
-         (c >= 'a' && c <= 'z') || (c >= '0' && c <= '9');
+constexpr std::array<uint8_t, 256> token_table = []() constexpr {
+  std::array<uint8_t, 256> result{};
+
+  for (uint8_t c : { '!' ,  '#' ,  '$' ,  '%' ,  '&' ,  '\'' ,
+          '*' ,  '+' ,  '\\' ,  '-' ,  '.' ,  '^' ,
+          '_' ,  '`' ,  '|' ,  '~'}) {
+    result[c] = true;
+  }
+
+  // (c >= 'A' && c <= 'Z')
+  for (uint8_t c = 65; c <= 90; c++) {
```

Comment:
can this be `c = 'A'` or something fpr easier readability?

Reply:
Yeah definitely. Updated the pull-request.